### ISSUE TITLE
T34 driver's view glitched after transitioning down to the leaning in, buttoned up view

### ISSUE THREAD

**Issue by handsm** [DarklightGames/DarkestHour#636]
Screenshots below show the normal view and then the glitched view.

Only affects multiplayer. View is ok when player first enters the driver position, but after mousewheeling up and back down again it often ends up looking like the 2nd picture. The camera position is about 10 units back (in the X axis) from where it should be.

![t34_driverview_normal](https://cloud.githubusercontent.com/assets/16405443/21996465/47da7036-dc22-11e6-883c-1a83d70d938a.jpg)

![t34_driverview_glitched](https://cloud.githubusercontent.com/assets/16405443/21996464/47d7fcfc-dc22-11e6-8ed7-3d91dc214882.jpg)

**Comment by handsm** [comment]
I came across a similar problem with the panther when creating a DH version of the animation file. It appeared that the driver's hatch closing anim often ended up one frame short of its final position, resulting in a glitched camera position like this. It was fine in RO and the anim files has simply been copied to a DH version and then edited, so all the anims were the same as in RO.

That was an annoying mystery and I fixed it with a workaround by editing the problem anim and simply adding an extra final, static frame as a throwaway, i.e. it didn't matter if that wasn't played. Was never fully happy with that.

Experimenting with this, it turns out the problem only happens when the server plays the same transition anim. In RO and earlier DH versions that did not usually happen, as the server didn't usually need to play such anims, seeing as they only had a visual effect. But we now play anims on the server, because they are required for the vastly improved vehicle occupant hit detection system. Hit detection for players in vehicle now operated the same as infantry players, so it detects hits on their body parts based on where they are, so it's essential the server moves the player into the same location & pose as the client, so hit detection is in sync.

Seemed bizarre that an anim playing independently on the server could screw up a client anim. Specifically, in this case, by stopping the client anim from reaching its final frame, leaving a camera bone in the wrong place. Looking at the settings concerning animations and replication, I saw there is an actor property called bReplicateAnimations, which is set to true in the ROWheeledVehicle class, from which all vehicles extend.

It appears this property should only affect replication of anim info called the SimAnim, which can be used with dump proxy actors, which vehicles certainly are not. SimAnim or bReplicateAnimations do not appear to be used anywhere in RO/DH and I can't see any reason why they should. The only anims played on a vehicle are the initial idle anim and any transition up/down anims, like the one in question. These are only played when entering state ViewTransition and that runs on both client and server independently. There really is no need or benefit of replicating anims for a vehicle.

But .... I tried removing the inherited bReplicateAnimations setting and it certainly fixes the problem. Totally illogical as far I can see or imagine, but must be something happening in native code. I've tested overriding this setting in the DHVehicle class, from which all out vehicles descend, and it solves the problem without creating any issues I can see.

This problem seems to show up with some RO tank anims, like the T34 and panther, because they have the player buttoning up and moving forward to the leaned in/periscope view in one fairly long anim, with the last frame having the camera bone surging forward through the visor/periscope to an external view. I think the problem, with anims ending up as frame short on the client, happens in other or all driver positions, but without being noticeable.

Anyway, happy to solve an old, niggling mystery, even if does pose a new, lesser one (as in: "why??"). And I'll put the old panther button up anim back one day, without the extra frame!
